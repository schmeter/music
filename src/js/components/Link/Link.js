import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';


const isExternal = (path) => /^https?:\/\/.+/.test(path);

class Link extends React.Component {
    render() {
        const { className, to, onClick, children } = this.props;
        const props = {
            className,
            onClick
        };
        const content = children || to;
        return isExternal(to) ? (
            <a
                target="_blank"
                href={to}
                {...props}
            >
                {content}
            </a>
        ) : (
            <NavLink
                exact
                activeClassName="active"
                to={to}
                {...props}
            >
                {content}
            </NavLink>
        );
    }
}

Link.propTypes = {
    to: PropTypes.string.isRequired,
    children: PropTypes.node,
    onClick: PropTypes.func,
    className: PropTypes.string
};

export default Link;
