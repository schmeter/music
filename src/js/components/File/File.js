import React, { useCallback } from 'react';
import PropTypes from 'prop-types';

import Link from '../Link';

const File = ({
    path,
    children,
    onClickFile,
}) => {
    const handleClickLink = useCallback(e => {
        e.preventDefault();
        onClickFile();
    }, [
        onClickFile,
    ]);

    return (
        <Link
            className="file"
            to={path}
            onClick={handleClickLink}
        >
            {children}
        </Link>
    );
};

File.propTypes = {
    path: PropTypes.string.isRequired,
    children: PropTypes.node.isRequired,
    onClickFile: PropTypes.func.isRequired,
};

export default File;
