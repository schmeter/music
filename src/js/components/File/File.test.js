import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import File from './File';
import Link from '../Link';

const mockedEvent = {
    preventDefault: jest.fn()
};

it('renders correctly', () => {
    const props = {
        path: 'http://test',
        onClickFile: jest.fn()
    };

    const component = shallow(<File {...props} >test</File>);

    expect(toJson(component)).toMatchSnapshot();

    component.find(Link).simulate('click', mockedEvent);
    expect(props.onClickFile).toHaveBeenCalled();
});
