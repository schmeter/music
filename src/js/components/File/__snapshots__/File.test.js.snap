// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File renders correctly 1`] = `
<Link
  className="file"
  onClick={[Function]}
  to="http://test"
>
  test
</Link>
`;
